OsmAnd/src/net/osmand/plus/FavouritesDbHelper.java

```diff
@@ -104,6 +111,96 @@ public void run() {
 		});
 	}
 
+	boolean hasWorkPoint() {
+		return hasPersonalPoint(WORK);
+	}
+
+	public boolean hasHomePoint() {
+		return hasPersonalPoint(HOME);
+	}
+
+	boolean hasParkingPoint() {
+		return hasPersonalPoint(PARKING);
+	}
+
+	private boolean hasPersonalPoint(String name) {
+		boolean hasPoint = false;
+		for (FavouritePoint fp : cachedFavoritePoints) {
+			if (fp.getName().equals(name) && fp.getCategory().equals(PERSONAL)) {
+				hasPoint = true;
+				break;
+			}
+		}
+		return hasPoint;
+	}
+
+	public FavouritePoint getWorkPoint() {
+		return getPersonalPoint(WORK);
+	}
+
+	public FavouritePoint getHomePoint() {
+		return getPersonalPoint(HOME);
+	}
+
+	public FavouritePoint getParkingPoint() {
+		return getPersonalPoint(PARKING);
+	}
+
+	LatLon getWorkPointLatLon() {
+		LatLon workPointLatLon;
+		if (hasWorkPoint()) {
+			workPointLatLon = new LatLon(getWorkPoint().getLatitude(), getWorkPoint().getLongitude());
+		} else {
+			workPointLatLon = new LatLon(0, 0);
+		}
+		return workPointLatLon;
+	}
+
+	LatLon getHomePointLatLon() {
+		LatLon homePointLatLon;
+		if (hasHomePoint()) {
+			homePointLatLon = new LatLon(getHomePoint().getLatitude(), getHomePoint().getLongitude());
+		} else {
+			homePointLatLon = new LatLon(0, 0);
+		}
+		return homePointLatLon;
+	}
+
+	LatLon getParkingPointLatLon() {
+		LatLon parkingPointLatLon;
+		if (hasParkingPoint()) {
+			parkingPointLatLon = new LatLon(getParkingPoint().getLatitude(), getParkingPoint().getLongitude());
+		} else {
+			parkingPointLatLon = new LatLon(0, 0);
+		}
+		return parkingPointLatLon;
+	}
+
+	PointDescription getHomePointDescription() {
+		return getHomePoint().getPointStringDescription();
+	}
+
+	PointDescription getWorkPointDescription() {
+		return getWorkPoint().getPointStringDescription();
+	}
+
+	PointDescription getParkingPointDescription() {
+		return getParkingPoint().getPointStringDescription();
+	}
+
+
+	private FavouritePoint getPersonalPoint(String name) {
```

Comment:
I suggest to store cachedHomePoint, cachedWorkPoint and cachedParkingPoint and not iterate all favorites.

Reply:
Yes. I think the same